ui/app/AppLayouts/Communities/views/PermissionsView.qml

```diff
@@ -87,14 +88,50 @@ ColumnLayout {
         ]
     }
 
+    Loader {
+        id: repeaterLoader
+        active: root.useRepeater
+        Layout.fillWidth: active
+        Layout.fillHeight: active
 
-    Repeater {
-        id: repeater
+        sourceComponent: ColumnLayout {
+            readonly property alias count: repeater.count
 
-        model: root.permissionsModel
+            width: root.viewWidth
+            spacing: 24
 
-        delegate: PermissionItem {
+            Repeater {
+                id: repeater
+                model: root.permissionsModel
+                delegate: permissionItemComponent
+            }
+        }
+    }
```

Comment:
This is the change that I'm less happy with.

I was able to convert this component to a ListView, and it worked well in the admin tab, however, when I checked the EditChannelPopup, it was all smooshed (no height). Also, it had a double scrollbar, which is not good.

I came up with this solution where if we are in the EditChannelPopup, we use the Repeater, and in the admin tab we use the ListView, but if you have an idea on how to make the ListView work in the popup, please let me know.

Reply:
Not good... you could use a ListView with expanded height (configurable), to avoid the double scrollbar. This double solution should be removed imho

Reply:
Ok, I was able to make it work!

Please review again. I also improved the speed even further by adding `prepareTokenModelForCommunityChat` that filters the permissions in Nim instead of doing it in the SortFilterProxyModel. The CreateChannelPopup opens way faster now

[edit-channel.webm](https://github.com/user-attachments/assets/1612caf5-82eb-4454-ac11-7a58517442cf)